lms/djangoapps/courseware/module_render.py

```diff
@@ -377,6 +458,13 @@ def handle_grade_event(block, event_type, event):
 
         dog_stats_api.increment("lms.courseware.question_answered", tags=tags)
 
+        # Fulfill any milestones for this content module now that all the data has been saved
+        _fulfill_content_milestones(
```

Comment:
For performance reasons, ideally we'd only call this if the graded item was within an entrance exam. Thus my suggestion to make is_entrance_exam be 'inherited metadata' so that the setting on the Chapter would auto-magically appear on all of the content subtree.